parser-typechecker/src/Unison/Codebase/SqliteCodebase/Conversions.hs

```diff
@@ -242,8 +242,7 @@ symbol2to1 :: V2.Symbol -> V1.Symbol
 symbol2to1 (V2.Symbol i t) = V1.Symbol i (Var.User t)
 
 symbol1to2 :: V1.Symbol -> V2.Symbol
-symbol1to2 (V1.Symbol i (Var.User t)) = V2.Symbol i t
-symbol1to2 x = error $ "unimplemented: symbol1to2 " ++ show x
+symbol1to2 (V1.Symbol i varType) = V2.Symbol i (Var.rawName varType)
```

Comment:
@aryairani @dolio okay yeah, this isn't quite the right fix, because the raw variable name may shadow or collide with another user-chosen name that's in the parent expression. In general, it's not safe to just rewrite a single variable using only local info like that, because of the possibility of collisions. You need to pass in something that is a closed term / type (no free vars) so the renaming of variables can be done in a way that avoids accidental capture.

If it's just a type signature you're tweaking, I think the correct fix will involve calling this function, passing it the full type:

https://github.com/unisonweb/unison/blob/477371ba97a019fe36294a76faed52190ef29d75/parser-typechecker/src/Unison/Typechecker/Context.hs#L274-L281

It will pick nice type variable names and avoid variable capture.

If you have a term with embedded annotations, then you might have to do something fancier to avoid capturing type variables from an outer scope. Hmm, thinking through this, I'm not sure it's possible for a type signature to refer to synthetic type variables in an outer scope. By design, there's no syntax for synthetic type variables, so the user can't write them in an explicit type signature. And the only way to capture a type variable from an outer scope is with an explicit type signature. So it seems that just recursively calling `removeSyntheticTypeVars` on any embedded type annotations in a term might be fine.

If it were necessary to do something fancier to deal with scoped type variable capture inside of terms, these functions do that properly and you could probably take inspiration from them and I'd be happy to pair.

https://github.com/unisonweb/unison/blob/98f2ed3b74995483091c6dac89d43d60c41ccaf0/unison-core/src/Unison/Term.hs#L295

https://github.com/unisonweb/unison/blob/98f2ed3b74995483091c6dac89d43d60c41ccaf0/unison-core/src/Unison/Term.hs#L311-L314

Reply:
This function is just about converting a single symbol from one format to another, and this PR is just about fixing a crash in it. 

What you're saying sounds to me like there's a bigger problem that's unrelated to this function or this PR.  We get the same bad pretty-term even with `--old-codebase`, which doesn't call this code.

So I think we should merge this and open a ticket for the thing I don't yet understand, and maybe @dolio could take point on? 🙏 

WDYT?

Reply:
I could back out or move the transcript, given that the output is wrong, but it's not related to this PR, so they could also be existing transcripts that are wrong for the same reason.

Reply:
@aryairani yeah, I'm okay with merging as is, just to avoid the crash, even though it's not correct, as long you create a separate ticket to track the proper fix.

It will only come up if someone picks one of those weird blackboard symbols for a type signature, which I think should be pretty rare.

Reply:
@pchiusano Our example case wasn't due to blackboard symbols, but due to pretty-printing a value that is a function that does pattern matching:
```
.> view List.map

  base.List.map : (a ->{𝕖} b) -> [a] ->{𝕖} [b]
  base.List.map f a =
    go i as acc =
      match List.at i as with
        None   -> acc
        Some a ->
          use Nat +
          go (i + 1) as (acc :+ f a)
    go 0 a []

.> display List.map

  go f i as acc =
    _pattern = List.at i as
    match _pattern with
      None           -> acc
      Some _pattern1 ->
        use Nat +
        go f (i + 1) as (acc :+ f _pattern)
  f a -> go f 0 a []
```